app/com/linkedin/drelephant/mapreduce/TaskLevelAggregatedMetrics.java

```diff
@@ -106,6 +106,9 @@ private void compute(MapReduceTaskData[] taskDatas, long containerSize, long ide
     }
 
     for (MapReduceTaskData taskData: taskDatas) {
+      if (!taskData.isTimeAndCounterDataPresent()) {
```

Comment:
No, it's not always true. If sampling is enabled, then it's only going to be true for sampled tasks. In `MapReduceFetcherHadoop2` class, `MapReduceTaskData` instance is created for all the tasks, but time and counter data is only stored for sampled tasks. Please look at the logic [here](https://github.com/linkedin/dr-elephant/blob/master/app/com/linkedin/drelephant/mapreduce/fetchers/MapReduceFetcherHadoop2.java#L315).
The unit test in this change will fail because NullPointerException without this check.